agent/app/agent.go

```diff
@@ -143,6 +143,12 @@ func newAgent(blackholeEC2Metadata bool, acceptInsecureCert *bool) (agent, error
 	seelog.Infof("Amazon ECS agent Version: %s, Commit: %s", version.Version, version.GitShortHash)
 	seelog.Debugf("Loaded config: %s", cfg.String())
 
+	if cfg.OnPrem.Enabled() {
```

Comment:
should we move this up to line 125 so that `ec2.NewEC2MetadataClient(nil)` is never called?

Reply:
then we wouldnt need to override inside of NewConfig either

Reply:
at line 125, the config has not been initialized yet so we don't know whether its on prem or not. from my testing, invoking ec2.NewEC2MetadataClient on-prem seems fine (didn't cause any error or delay)